index.html

```diff
@@ -218,6 +237,400 @@ <h2>
+            context=] can be navigated . All navigations within the application
+            to a URL [=within home tab scope=] will be performed in the [=home
+            tab context=], and all navigations to a URL [=outside of home tab
+            scope=] will be performed in a different [=application context=]
+            (i.e., another tab), creating one if necessary. The application's
+            [=start URL=] is always part of the home tab scope.
+          </p>
+          <p>
+            For the purpose of matching against the [=start URL=], the
+            [=URL/fragment=] is ignored, but the [=URL/query=] must match
+            exactly.
+          </p>
+          <p>
+            If the application does not [=has a home tab|have a home tab=],
+            then the home tab scope is the empty set.
+          </p>
+        </aside>
+        <section>
+          <h3>
+            Every window has a home tab
+          </h3>
+          <p>
+            If the application [=has a home tab=], whenever a new application
+            window is created (for example when launching the application, or
+            when moving a tab to a new window), the user agent MUST create a
+            new [=home tab context=] in that window. A newly created [=home tab
+            context=] SHOULD be navigated to the [=start URL=], which by
+            definition is [=within home tab scope=].
+          </p>
+        </section>
+        <section>
+          <h3>
+            Navigations concerning the home tab scope
+          </h3>
+          <p>
+            If navigating from the [=home tab context=] to a [=URL=] |url:URL|
+            that is [=outside of home tab scope=], the following steps are run:
+          </p>
+          <ol>
+            <li>Create a new [=application context=] in the current window and
+            focus it.
+            </li>
+            <li>Navigate it to |url|.
+            </li>
+          </ol>
+          <p>
+            If navigating to a [=URL=] |url:URL| [=within home tab scope=], the
+            following steps are run:
+          </p>
+          <ol>
+            <li>Focus the [=home tab context=].
+            </li>
+            <li>Navigate it to |url|.
+            </li>
+          </ol>
+          <aside class="note">
+            Navigation from the home tab to a URL [=outside of home tab scope=]
+            will open a new regular tab, leaving the existing document open in
+            the home tab. Similarly, navigation from a regular tab to a URL
+            [=within home tab scope=] will open in the home tab, leaving the
+            existing document open in the tab that initiated the navigation.
+            Developers should ensure that in both of these cases, the document
+            left behind does not end up in an unexpected state.
+          </aside>
+        </section>
+        <section class="informative">
+          <h3>
+            Home tab invariants
```

Comment:
These invariants are likely 'cleaned up' or explained better by my above suggestion about exactly where the 'home tab context' is chosen to be used for a navigation. Then this is more of an informative note / explanation of the behavior, and the edge cases are:
- window.opener (actually this might not be a huge deal - you can just choose the home browsing context for the navigation & allow that to be part of the browsing group the link came from)
- frame name is specified, which already exists
- self navigation (kinda the same as frame name)
- the history modification API as you mentioned
- others?

Reply:
This section is intended to be informative. I intend for this to be (non-normatively) the overriding philosophy of how tabs work, while the normative text makes sure that it happens.

I'll have to think about window.opener and frames which I hadn't previously considered. However I think frames don't matter because this only applies to top-level browsing contexts.

Reply:
We had an offline discussion about this. The summary is:

1. The paragraph that explains to developers of single-page applications that they should do real navigations in and out of home scope, is to be reworded so it's clear that it's giving developers advice, rather than highlighting an issue.
2. The "navigations concerning the home tab scope" should be reworked (and we workshopped a sketch of how the algorithm should look, which is as follows):

> When [choosing a navigable for a navigation ](https://html.spec.whatwg.org/multipage/document-sequences.html#the-rules-for-choosing-a-navigable) from the [home tab context](https://pr-preview.s3.amazonaws.com/mgiuca/manifest-incubations/pull/95.html#dfn-home-tab-context) to a [URL](https://url.spec.whatwg.org/#concept-url) url that is [outside of home tab scope](https://pr-preview.s3.amazonaws.com/mgiuca/manifest-incubations/pull/95.html#dfn-outside-of-home-tab-scope), the following steps are run:
>
> 1. Behave as `target=_blank`, with _noopener_ and _noreferrer_ specified, creating a new top-level traversable _t_.
t is created in the current window and has the manifest applied.
> 2. Focus _t_.
>
> When [choosing a navigable for a navigation](https://html.spec.whatwg.org/multipage/document-sequences.html#the-rules-for-choosing-a-navigable), if the [URL](https://url.spec.whatwg.org/#concept-url) url is [within home tab scope](https://pr-preview.s3.amazonaws.com/mgiuca/manifest-incubations/pull/95.html#dfn-within-home-tab-scope), the following steps are run:
>
> 1. Let _t_ be the home tab context’s traversable associated with the current window.
> 2. Override the [algorithm](https://html.spec.whatwg.org/multipage/document-sequences.html#the-rules-for-choosing-a-navigable) to return _t_.
> 3. Focus _t_.

Reply:
@dmurph 

A couple of issues arose trying to implement this:

1. It turns out that "[choosing a navigable](https://html.spec.whatwg.org/multipage/document-sequences.html#the-rules-for-choosing-a-navigable)" is not exported, so for whatever reason, we are not supposed to reference it from outside of HTML. I'm not sure if that means we just shouldn't, or should try and export it, but to avoid shaving that yak first, I think I will just hand-wave it instead of directly referencing this algorithm.
2. We don't actually want to always override the algorithm, only when target=_self. (So for instance, if you click a link with target=<name> then it should still open in the named target.) However, conversely, we want to capture navigations that target this tab, even if they didn't come from this tab (for example, if a child frame does a navigation with target=_parent, into a home URL, we want to capture that into the home tab). So actually I think hooking into the "choosing a navigation" is not the right place. We should let that algorithm run its course, and then _after_ a navigable has been chosen, if it is one of our tabs, override the navigable that actually gets navigated.

I think on account of both of these, it makes more sense to go back to hooking into the algorithm at the "navigate" step. I'll rework it and send it for review.

Reply:
I've rewritten it to capture the spirit of what we talked about this morning.

I am using "choose a navigable" to get the target=_blank logic, with a red box noting that I can't actually access it (#98).

But primarily, this is now about intercepting "navigation" and doing an "instead of navigating X, navigate Y" type of logic. I think that is more precisely what we want, because then we will intercept if _anything_ in the system tries to navigate one of our tabs.

Reply:
LGTM